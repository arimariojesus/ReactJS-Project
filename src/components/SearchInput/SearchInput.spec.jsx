import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { SearchInput } from '.';

describe('<SearchInput />', () => {
  it('should have a value of value', () => {
    const fn = jest.fn();
    render(<SearchInput value={'testando'} onChange={fn} />);

    const input = screen.getByPlaceholderText(/type your search/i);
    expect(input.value).toBe('testando');
  });

  it('should call handleChange function on each key pressed', () => {
    const fn = jest.fn();
    render(<SearchInput onChange={fn} />);

    const input = screen.getByPlaceholderText(/type your search/i);
    const value = 'any value';

    userEvent.type(input, value);

    setTimeout(() => {
      expect(input.value).toBe(value);
      expect(fn).toHaveBeenCalledTimes(1);
    }, 500);
  });

  it('should call handleChange function on each key pressed', () => {
    const fn = jest.fn();
    const { container } = render(<SearchInput onChange={fn} />);
    expect(container).toMatchSnapshot();
  });
});