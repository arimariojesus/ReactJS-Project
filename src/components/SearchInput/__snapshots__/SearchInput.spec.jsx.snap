// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> should call handleChange function on each key pressed 1`] = `
<div>
  <input
    class="search-input"
    placeholder="Type your search"
    type="search"
    value=""
  />
</div>
`;
